windows/power_operation_scripts/set_power_script.yml

```diff
@@ -0,0 +1,152 @@
+
+- name: "Pause 10 seconds to wait configure take effect"
+  ansible.builtin.pause:
+    seconds: 10
+
+- name: "Shutdown and then power on guest OS for {{ power_script_type }} {{ power_script_op }}"
+  when: power_script_op in ['power', 'shutdown']
+  block:
+    - name: "Shutdown VM for {{ power_script_type }} {{ power_script_op }}"
+      include_tasks: ../../common/vm_set_power_state.yml
+      vars:
+        vm_power_state_set: "shutdown-guest"
+
+    - name: "Power on VM for {{ power_script_type }} {{ power_script_op }}"
+      include_tasks: ../../common/vm_set_power_state.yml
+      vars:
+        vm_power_state_set: "powered-on"
+
+- name: "Suspend and resume VM for {{ power_script_type }} {{ power_script_op }}"
+  when: power_script_op in ['suspend', 'resume']
+  block:
+    - name: "Suspend the VM for {{ power_script_type }} {{ power_script_op }}"
+      include_tasks: ../../common/vm_set_power_state.yml
+      vars:
+        vm_power_state_set: 'suspended'
+
+    - name: "Resume VM for {{ power_script_type }} {{ power_script_op }}"
+      include_tasks: ../../common/vm_set_power_state.yml
+      vars:
+        vm_power_state_set: "powered-on"
+        force: true
+
+- name: "Update inventory"
+  include_tasks: ../utils/win_update_inventory.yml
+
+- name: "Wait for VMware Tools service is running"
+  include_tasks: ../utils/win_wait_service_status.yml
+  vars:
+    win_service_name: 'VMTools'
+
+- name: "Get content of file tools.conf after {{ power_script_type }} {{ power_script_op }}"
+  include_tasks: ../utils/win_execute_cmd.yml
+  vars:
+    win_powershell_cmd: "& type '{{ vmtools_config_file }}'"
+
+- name: "Check the script file running as expected"
+  when:
+    - script_tag_path is defined and script_tag_path
+    - power_script_type == "custom"
+    - power_script_op != "suspend"
+  block:
+    - name: "Get the status of file {{ script_tag_path }} for {{ power_script_type }} {{ power_script_op }}"
+      ansible.windows.win_stat:
+        path: "{{ script_tag_path }}"
+      delegate_to: "{{ vm_guest_ip }}"
+      register: tag_path_exist_result
+      until:
+        - tag_path_exist_result.stat.exists is defined
+        - tag_path_exist_result.stat.exists
+      retries: 30
+      delay: 5
+
+    - name: "Check the searching result"
+      ansible.builtin.assert:
+        that:
+          - tag_path_exist_result.stat.exists is defined
+          - tag_path_exist_result.stat.exists
+        fail_msg: "Failed to find '{{ script_tag_path }}' for {{ power_script_type }} {{ power_script_op }}."
+        success_msg: "Successfully found the '{{ script_tag_path }}' for {{ power_script_type }} {{ power_script_op }}."
+
+- name: "Check the log file"
+  when: power_script_op != "suspend"
+  block:
+    - name: "Search the log file {{ script_file_path }} in {{ guest_os_ansible_distribution }}"
+      ansible.windows.win_shell: "& findstr '.*{{ script_file_path | win_basename }}' *.log"
```

Comment:
Can here '*.log' be changed to a specific file name?

Reply:
Not sure which log file we can search the specific string successfully.